arviz/plots/backends/matplotlib/posteriorplot.py

```diff
@@ -158,15 +165,20 @@ def display_ref_val():
             val,
             format_as_percent(greater_than_ref_probability, 1),
         )
-        ax.axvline(val, ymin=0.05, ymax=0.75, color="C1", lw=linewidth, alpha=0.65)
+        ref_val_props = {
+            "color": vectorized_to_hex("C1")
+            if ref_val_color is None
+            else vectorized_to_hex(ref_val_color)
+        }
+        ax.axvline(val, ymin=0.05, ymax=0.75, lw=linewidth, alpha=0.65, **ref_val_props)
```

Comment:
you can pass `color` as a kwarg directly to `axvline` no need to create the dict, as it is only one kwarg it does not make a difference (also see other comments on removing the if else)